s2a.go

```diff
@@ -134,20 +144,38 @@ func NewServerCreds(opts *ServerOptions) (credentials.TransportCredentials, erro
 			info: &credentials.ProtocolInfo{
 				SecurityProtocol: s2aSecurityProtocol,
 			},
-			minTLSVersion: commonpb.TLSVersion_TLS1_3,
-			maxTLSVersion: commonpb.TLSVersion_TLS1_3,
-			tlsCiphersuites: []commonpb.Ciphersuite{
-				commonpb.Ciphersuite_AES_128_GCM_SHA256,
-				commonpb.Ciphersuite_AES_256_GCM_SHA384,
-				commonpb.Ciphersuite_CHACHA20_POLY1305_SHA256,
+			minTLSVersion: commonpbv1.TLSVersion_TLS1_3,
+			maxTLSVersion: commonpbv1.TLSVersion_TLS1_3,
+			tlsCiphersuites: []commonpbv1.Ciphersuite{
+				commonpbv1.Ciphersuite_AES_128_GCM_SHA256,
+				commonpbv1.Ciphersuite_AES_256_GCM_SHA384,
+				commonpbv1.Ciphersuite_CHACHA20_POLY1305_SHA256,
 			},
 			localIdentities: localIdentities,
 			isClient:        false,
 			s2aAddr:         opts.S2AAddress,
 		}, nil
 	}
 	verificationMode := getVerificationMode(opts.VerificationMode)
-	return v2.NewServerCreds(opts.S2AAddress, opts.TransportCreds, localIdentities, verificationMode, opts.getS2AStream)
+	var marshalledLocalIdentities [][]byte
+	for _, localIdentity := range localIdentities {
+		marshalledLocalIdentity, err := proto.Marshal(localIdentity)
+		if err != nil {
+			return nil, err
+		}
+		marshalledLocalIdentities = append(marshalledLocalIdentities, marshalledLocalIdentity)
+	}
+	var v2LocalIdentities []*commonpb.Identity
+	for _, marshalledLocalIdentity := range marshalledLocalIdentities {
+		v2LocalIdentity := &commonpb.Identity{}
+		err := proto.Unmarshal(marshalledLocalIdentity, v2LocalIdentity)
+		if err != nil {
+			return nil, err
+		}
+		v2LocalIdentities = append(v2LocalIdentities, v2LocalIdentity)
+	}
+
```

Comment:
Do you think it would be better to have a separate toProtoIdentity() function for the v2 identity? instead of doing the marshal & un-marshal conversion.

My understanding is that at some point in the future, the v1/legacy code can be removed, and the marshal/unmarshal logic will need to be replaced anyway. wdyt?

Reply:
My understanding is the same as yours. I have added a new ```toV2ProtoIdentity``` function in s2a_options.go, and replaced the marshal/unmarshal logic with this.